### PR TITLE
refactor: anchor existence check and add tests for broken anchor links

### PR DESCRIPTION
**Pull request by SteveJonesDev**
This pull request enhances the functionality of the `anchor_exists` check by adding support for validating anchor links that target elements with a `name` attribute, in addition to those with an `id`. It also introduces comprehensive Jest tests to validate the behavior of the updated check across various scenarios, including edge cases.

### Enhancements to `anchor_exists` check:
* Updated `src/pageScanner/checks/anchor-exists.js` to validate anchor links by first checking for an element with a matching `id` and, if not found, looking for an anchor element with a matching `name` attribute.

### Addition of Jest tests:
* Added `tests/jest/rules/brokenAnchorLink.test.js` to cover a wide range of scenarios for anchor link validation, including:
  - Links pointing to non-existent `id` or `name` attributes.
  - Links with valid targets, including elements with `id` or `name` attributes.
  - Edge cases such as empty fragments (`href="#"`), links with `role="button"`, and anchor names containing special characters.

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved detection of anchor links by ensuring both ID and named anchor targets are recognized, reducing false positives for broken anchor links.

- **Tests**
	- Added comprehensive tests to verify correct identification of broken and valid anchor links, including edge cases and special scenarios.

### CODE REVIEW

File: src/pageScanner/checks/anchor-exists.js [hunk 1]
```diff
@@ -8,6 +8,20 @@
 export default {
 	id: 'anchor_exists',
 	evaluate: ( node ) => {
-		return document.querySelector( node.getAttribute( 'href' ) ) !== null;
+		const href = node.getAttribute( 'href' );
+
+		// First try the standard CSS selector approach (finds elements with matching IDs)
+		if ( document.querySelector( href ) !== null ) {
+			return true;
+		}
```
[the remark this answers is not in the document]
Comment (SteveJonesDev) [reply src/pageScanner/checks/anchor-exists.js:16]: This is the selector: 'a[href^="#"]:not([href="#"]):not([role="button"])',

[the remark this answers is not in the document]
Comment (pattonwebz) [reply src/pageScanner/checks/anchor-exists.js:16]: @coderabbitai draft the patch
